security-proxy/src/main/java/org/georchestra/security/UserDetailsJSONRequestHeaderProvider.java

```diff
@@ -0,0 +1,264 @@
+import lombok.Setter;
+
+/**
+ * Adds a {@code sec-user} header with the full {@link GeorchestraUser} obtained
+ * from {@link AccountsService} as a Base64 encoded JSON representation.
+ * <p>
+ * This header provider is disabled by default, and reads which target request
+ * services to add the header for from {@code headers-mapping.properties} just
+ * like {@link LdapUserDetailsRequestHeaderProvider}.
+ * <p>
+ * Adding the {@code sec-user} header by default to all services can be enabled
+ * in through {@code send-json-sec-user=true}.
+ * <p>
+ * Individual service names for which to add the {@code sec-user} request header
+ * must be configured as {@code <service-name>.send-json-sec-user=true}. For
+ * example:
+ * 
+ * <pre>
+ * <code>
+ * geonetwork.send-json-sec-user=true
+ * datafeeder.send-json-sec-user=true 
+ * </code>
+ * </pre>
+ * 
+ * If this header provider is enabled globally, though, it can be disabled for
+ * any specific service by setting the service specific config property to
+ * {@code false}, as in {@code <service-name>.send-json-sec-user=false}.
+ */
+public class UserDetailsJSONRequestHeaderProvider extends HeaderProvider {
+
+    static final String CONFIG_PROPERTY = "send-json-sec-user";
+    private static final String HEADER_NAME = "sec-user";
+
+    /** Key used for lookup of whether this header provider is globally enabled **/
+    private static final String GLOBAL_KEY = "global";
+
+    private PropertyResolver env;
+
+    @Autowired
+    @Setter(value = AccessLevel.PACKAGE)
+    private AccountsService lookupService;
+
+    private @Autowired GeorchestraConfiguration georchestraConfiguration;
+
+    /**
+     * Encoder to create the JSON String value for a {@link GeorchestraUser}
+     * obtained from {@link AccountsService}
+     */
+    private ObjectMapper encoder;
+
+    /**
+     * Cache of Base64-JSON encoded {@link GeorchestraUser user} by username.
+     * <p>
+     * Entries should expire in a long enough time to avoid concurrent requests
+     * flooding the LDAP server with requests, but short enough to allow propagating
+     * changes down to proxied services asap.
+     */
+    private Cache<String, String> cache;
+
+    /**
+     * Individual service names for which to add the {@code sec-user} request
+     * header. Must be configured in {@code headers-mapping.properties} as
+     * {@code <service-name>.send-json-sec-user=true}, for example:
+     * 
+     * <pre>
+     * <code>
+     * geonetwork.send-json-sec-user=true
+     * datafeeder.send-json-sec-user=true 
+     * </code>
+     * </pre>
+     */
+    private Map<String, Boolean> enabledServices = Collections.singletonMap(GLOBAL_KEY, Boolean.FALSE);
+
+    /**
+     * Cache time-to-live in milliseconds, can be given through an external
+     * configuration property {@code security-proxy.ldap.cache.ttl}
+     */
+    private static final long DEFAULT_CACHE_TTL = 2000;
+
+    public UserDetailsJSONRequestHeaderProvider() {
```

Comment:
Checking the maximum size for http headers (not clear if it is for a single one or for multiple ones), it depends across webserver implementations:
https://www.tutorialspoint.com/What-is-the-maximum-size-of-HTTP-header-values
But we are quite limited (min 8KB), so we won't be able to transmit the jpegPhoto attribute using this one ;-)

Reply:
yeah, but a user won't come near close to that, especially now that the roles and organization are just strings, not the full objects. The jpegPhoto is not included in the `GeorchestraUser` class either, and IMHO it shouldn't. If it comes to requiring it, we should add a method to `UsersApi` to fetch a users picture instead.